mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageIdMapperTest.java

```diff
@@ -773,6 +773,67 @@ public void countUnseenMessageShouldTakeCareOfMessagesMarkedAsUnread() throws Ex
     }
 
     @ContractTest
+    public void setFlagsShouldNotUpdateModSeqWhenNoop() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        long modSeq = mapperProvider.generateModSeq(benwaInboxMailbox);
+        message1.setModSeq(modSeq);
+        message1.setFlags(new Flags(Flag.SEEN));
+        sut.save(message1);
+
+        sut.setFlags(message1.getMessageId(),
+            ImmutableList.of(message1.getMailboxId()),
+            new Flags(Flag.SEEN),
+            FlagsUpdateMode.ADD);
```

Comment:
There are three possible values for FlagsUpdateMode (enum)
- REPLACE: replace current flags list by the flags list provided
- ADD: add flags provided to the current flags list
- REMOVE: remove flags provided to the current flags list